lib/bolt/cli.rb

```diff
@@ -624,9 +618,9 @@ def migrate_group(group)
         group['targets'] = targets
       end
       (group['groups'] || []).each do |subgroup|
-        migrate_group(subgroup)
+        migrated = migrate_group(subgroup)
```

Comment:
This should be `||=`. `migrated` is a latch that ought to only ever be switched in one direction. If _at any point_ we find a `nodes` key, then we should set `migrated` to true, but we should under no circumstances ever set it back to false.

Currently, this will report no changes in this case:

```yaml
nodes:
  - foo.example.com
  - bar.example.com
groups:
  - name: somegroup
```

Admittedly, there's not a lot of reason to have a leaf group without a `nodes` key, but it's technically allowed.